### DISCUSSION TITLE
Use React Hooks in Custom Functions?

### DISCUSSION THREAD

**Discussion by zackrw** [handsontable/hyperformula#1341] [API Questions]
For example, if I have a useUser() hook, I want to register a custom function that can access that data. Might that be possible?

**Comment by sequba** [comment]
Hi @zackrw, thanks for reaching out.

It should be possible to use react hooks inside a custom function, but I haven't tried it, so I cannot be sure. If you encounter any problems with that, feel free to share your code and we'll do our best to help you.

Note, however, that asynchronous custom functions are not supported at the current stage of development. Some discussions about that:
- https://github.com/handsontable/hyperformula/issues/9#issuecomment-1029829216
- https://github.com/handsontable/hyperformula/issues/1119#issuecomment-1340998018
- https://github.com/handsontable/hyperformula/discussions/808
